### PR TITLE
FIX: Do not display the color scheme ID in interface dropdown

### PR DESCRIPTION
**Pull request by pmusaraj**
When a theme's default color scheme is not marked as user selectable, we were outputting the numeric ID in the UI. This outputs "Theme default" instead.

### CODE REVIEW

**Comment by jjaffeux** [comment]
Can't we fix this server side ?

File: app/assets/javascripts/discourse/tests/acceptance/user-preferences-interface-test.js [hunk 1]
```diff
@@ -189,6 +189,37 @@ acceptance(
...
+        selectKit(".light-color-scheme .select-kit").header().value(),
+        null
+      );
+      assert.equal(
+        selectKit(".light-color-scheme .select-kit").header().label(),
+        I18n.t("user.color_schemes.default_description")
+      );
+
+      await selectKit(".light-color-scheme .select-kit").expand();
+      assert.equal(count(".light-color-scheme .select-kit .select-kit-row"), 1);
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/tests/acceptance/user-preferences-interface-test.js:218]: you have a helper for rows:  `dropdownObject.rows().length`

File: app/assets/javascripts/discourse/tests/acceptance/user-preferences-interface-test.js [hunk 2]
```diff
@@ -189,6 +189,37 @@ acceptance(
...
+        { theme_id: 1, name: "A Theme", color_scheme_id: 2, default: true },
+      ]);
+
+      site.set("user_color_schemes", [{ id: 3, name: "A Color Scheme" }]);
+
+      await visit("/u/eviltrout/preferences/interface");
+
+      assert.ok(exists(".light-color-scheme"), "has regular dropdown");
+      assert.equal(
+        selectKit(".light-color-scheme .select-kit").header().value(),
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/tests/acceptance/user-preferences-interface-test.js:209]: I would recommend to hold the dropdown object in a var so you just reuse it

File: app/assets/javascripts/discourse/app/controllers/preferences/interface.js [hunk 3]
```diff
@@ -238,7 +238,27 @@ export default Controller.extend({
       return value;
     },
     get() {
-      return this.session.userColorSchemeId;
+      if (!this.session.userColorSchemeId) {
+        return;
+      }
+
+      const defaultTheme = this.site
+        .get("user_themes")
```
Comment (jjaffeux) [inline app/assets/javascripts/discourse/app/controllers/preferences/interface.js:246]: do we need this get ?

**Comment by pmusaraj** [comment]
> Can't we fix this server side ?

Yes, we can fix this server-side, I tried that initially but doing so added add one more query when bootstrapping (to pull the default theme's color scheme ID), and given that's a critical path, I opted to fix it client-side using the already available data under `user_themes`.

**Review by jjaffeux — APPROVED** [review]

**Pull request merged**